### PR TITLE
[LF-2836] [LF-2826] download a document from the documents home and read only page

### PR DESCRIPTION
**Pull request by anant00729**
To test: 

[LF-2826]
1. Go to the documents listing page.  
2. add a document 
3. check on the screen if you can find the download option.
4. click on the download button to see the uploaded file.

[LF-2836]
1. After adding the document. click on the document tile to go to the document details page.
2. On the preview you can find a download button, try clicking on it and check if the document is downloading.

For more details:
https://lite-farm.atlassian.net/browse/LF-2826
https://lite-farm.atlassian.net/browse/LF-2836

[LF-2826]: https://lite-farm.atlassian.net/browse/LF-2826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LF-2836]: https://lite-farm.atlassian.net/browse/LF-2836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by wasswarichard — APPROVED** [review]

**Comment by Duncan-Brain** [comment]
@anant00729 Is this something that is only testable on beta? I cannot really tell if this will work I am not familiar enough with the document and image storage server. 

But I would ask for a minor change that either: 

a) Instead of creating a duplicate of ImageWithAuthentication, we make it more generic to support both cases since it is very similar code

or

b) We edit the variable names to be document related instead of image related in DocumentWithAuthentication.

**Comment by anant00729** [comment]
@Duncan-Brain . I have made the requested changes. Please take a look. Thanks

File: packages/webapp/src/containers/MediaWithAuthentication/index.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  useEffect(() => {
+    const config = {
+      headers: {
+        Authorization: 'Bearer ' + localStorage.getItem('farm_token'),
+      },
+      responseType: 'arraybuffer',
+      method: 'GET',
+    };
+    let subscribed;
+    const getImageSrc = async () => {
```
Comment (Duncan-Brain) [inline packages/webapp/src/containers/MediaWithAuthentication/index.jsx:21]: @anant00729 should this be getMediaSrc?

**Comment by Duncan-Brain** [comment]
@anant00729 I added a few more line specific questions but otherwise looks good to me. 

Is there a way I can test this locally? I cannot currently test the download feature with my local config.

**Comment by anant00729** [comment]
@Duncan-Brain Here is the video to show how it works
[screen-capture (2).webm](https://user-images.githubusercontent.com/20675885/199776603-c1f584f4-9f2b-4c5e-94af-906846e13f62.webm)

**Review by Duncan-Brain — APPROVED** [review]
Thanks for sharing the video!

**Review by wasswarichard — APPROVED** [review]

**Pull request merged**